x-pack/plugins/security_solution/public/flyout/shared/hooks/use_on_close.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { useWhichFlyout } from '../../document_details/shared/hooks/use_which_flyout';
+import { Flyouts } from '../../document_details/shared/constants/flyouts';
+import { SECURITY_SOLUTION_ON_CLOSE_EVENT, TIMELINE_ON_CLOSE_EVENT } from '../..';
+
+export interface UseOnCloseParams {
+  /**
+   * Function to call when the event is dispatched
+   */
+  callback: (id: string) => void;
+}
+
+/**
+ * Hook to abstract the logic of listening to the onClose event for the Security Solution application.
+ * The kbn-expandable-flyout package provides the onClose callback, but has there are only 2 instances of the expandable flyout in Security Solution (normal and timeline)
+ * we need a way to propagate the onClose event to all other components.
+ * 2 event names are available, we pick the correct one depending on which flyout is open (if the timeline flyout is open, it is always on top, so we choose that one).
+ */
+export const useOnClose = ({ callback }: UseOnCloseParams): void => {
+  const flyout = useWhichFlyout();
+
+  const eventName =
```

Comment:
Can/Should there be a generic flyout event? or we need to create for everyone.

Reply:
I'm not sure I understand the question... Currently we only have 2 expandable flyouts for the entire Security Solution application, and this hook is only usable with the Security Solution application.
What did you mean by generic flyout event?

Reply:
yes.. sorry I thought, Expandable flyout will globally emit one even for both alert and timeline flyout. Please ignore my comment.